lib/plug_rails_cookie_session_store.ex

```diff
@@ -81,14 +91,21 @@ defmodule PlugRailsCookieSessionStore do
     key_opts = opts.key_opts
     cookie = cookie |> URI.decode_www_form()
 
-    if key = opts.encryption_salt do
-      MessageEncryptor.verify_and_decrypt(
+    if opts.use_authenticated_encryption do
+      MessageEncryptor.authenticate_and_decrypt(
         cookie,
-        derive(conn, key, key_opts),
-        derive(conn, opts.signing_salt, key_opts)
+        derive(conn, opts.authenticated_encryption_salt, key_opts |> Keyword.put(:key_digest, :sha1))
```

Comment:
```suggestion
derive(conn, opts.authenticated_encryption_salt, key_opts |> Keyword.put(:digest, :sha))
```
I've tried use this branch with real Rails cookie. I've discovered that secret is generated differently. It was because `Plug.Crypto.KeyGenerator` uses key `:digest` instead of `:key_digest` and because `sha1` algorithm has to be represented as just `:sha` (without the "1").
Not sure if it changed recently.
I am on Erlang 23.2.5